proto/streamlit/proto/NewSession.proto

```diff
@@ -69,6 +69,10 @@ message NewSession {
 
   // Hash of the main script running (ie streamlit run main_script.py)
   string main_script_hash = 11;
+
+  // Any query string that has been set by the backend for this session; should
+  // not include any embed= or embed_theme= keys.
+  string query_string = 12;
```

Comment:
may have missed this -- are we using this field?

Reply:
You are correct about `PageLink`, however you need this field for `switch_page()`, which spawns a new re-run on the backend. Including the `query_string` field in the `NewSession` is the best and (currently) only way for it to tell the front-end the query string has changed.